data/vector.go

```diff
@@ -226,7 +229,7 @@ const (
 )
 
 // MarshalJSON marshals the enum as a quoted json string
-func (p *FieldType) MarshalJSON() ([]byte, error) {
+func (p FieldType) MarshalJSON() ([]byte, error) {
 	buffer := bytes.NewBufferString(`"`)
 	buffer.WriteString(p.ItemTypeString())
 	buffer.WriteString(`"`)
```

Comment:
You could write `return json.Marshal(p.ItemTypeString())` here, but maybe you aimed for performance  - if yes - then right, this code performs a bit better than `json.Marshal` (we can also use byte buffer pool here in the future to remove buffer allocation and improve encoding speed slightly).